here-naksha-app-service/src/main/java/com/here/naksha/app/service/http/tasks/NoElementsStrategy.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2017-2023 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+package com.here.naksha.app.service.http.tasks;
+
+import com.here.naksha.lib.core.models.XyzError;
+
+public enum NoElementsStrategy {
+  FAIL_ON_NO_ELEMENTS(
```

Comment:
I think the error names are correct for what they represent though. These errors will only be sent when a NoCursor or NoSuchElementException is caught, so the name of "NoElementsStrategy" and "_ON_NO_ELEMENTS" suffix reflect that. Perhaps changing the first one to UNEXPECTED_RESULT_ON_NO_ELEMENTS would be more appropriate, if needed?

Reply:
I only intended to differentiate between FAIL_xxx and NOT_FOUND_xxx , because to me both of them are kind of failures. But, this is not that important. We can change in future as well, if it becomes confusing. Not a problem. Anything is fine.